File: reverseWords.java
// Approach 1- Two Pointers 
class Solution { // TC O(n) SC O(1)
    public String reverseWords(String s) {
        int lastSpaceIndex = -1;
        char[] chArray = s.toCharArray();
        int len = s.length();
        for (int strIndex = 0; strIndex <= len; strIndex++) {
            if (strIndex == len || chArray[strIndex] == ' ') {
                int startIndex = lastSpaceIndex + 1;
                int endIndex = strIndex - 1;
                while (startIndex < endIndex) {
                    char temp = chArray[startIndex];
                    chArray[startIndex] = chArray[endIndex];
                    chArray[endIndex] = temp;
                    startIndex++;
                    endIndex--;
                }
                lastSpaceIndex = strIndex;
            }
        }
        return new String(chArray);
    }

}

class reverseWords{
    public static void main(String[] args) {
        String x= "Lets take leetcode contest"; 
        Solution s= new Solution();
        System.out.println(s.reverseWords(x));
    }
}

File: reverseWords.py
def reverseWords(s):
	return " ".join([word[::-1] for word in s.split(" ")])

def reverseWords1(s):
	s1=s[::-1]
	ans=s1.split(" ")
	ans1=ans[::-1]
	return " ".join(ans1)

print(reverseWords("Lets take leetcode test"))
print(reverseWords1("Lets take leetcode test"))

